qiskit_experiments/library/randomized_benchmarking/rb_analysis.py

```diff
@@ -134,20 +134,32 @@ def _initial_guess(
         return opt
 
     def _format_data(self, data: curve.CurveData) -> curve.CurveData:
-        """Take average over the same x values."""
-        mean_data_index, mean_x, mean_y, mean_e = multi_mean_xy_data(
+        """Data format with averaging with sampling strategy."""
+        # take average over the same x value by regenerating sigma from variance of y values
+        series, xdata, ydata, sigma, shots = multi_mean_xy_data(
             series=data.data_index,
             xdata=data.x,
             ydata=data.y,
             sigma=data.y_err,
+            shots=data.shots,
             method="sample",
         )
+
+        # sort by x value in ascending order
+        series, xdata, ydata, sigma = data_sort(
+            series=series,
+            xdata=xdata,
+            ydata=ydata,
+            sigma=sigma,
```

Comment:
I think `data_sort` should also return `shots`. Looks like there's a bug here, where the array of shots is not permuted, along with the other arrays, during the sort.